source/engine/Orders.h

```diff
@@ -0,0 +1,70 @@
+/* Orders.h
+Copyright (c) 2024 by TomGoodIdea
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "../Point.h"
+
+#include <memory>
+
+class Minable;
+class Ship;
+class System;
+
+
+
+// This is a base class for handling ship orders. It contains members common
+// for OrderSet and OrderSingle: information about targets and the list
+// of possible order types.
+class Orders {
+public:
+	enum Types {
```

Comment:
```suggestion
	enum class Types : int {
```
This should be an `enum class` so it doesn't pollute any namespaces.

Reply:
Doesn't it already require to use the Orders namespace outside the order classes?

Reply:
Yes, but the style guide is to use enum classes by default.

Reply:
In general, yes (and it should probably be added to the written version of the guide after the migration). In this case, the enum is a core part of the class, and I wouldn't call it pollution, but code simplification ~~and you're the one who uses auto for everything smh~~.
That said, I'm fine with changing it if you still think it would be better.

Reply:
Hmm
I think I managed to confuse myself with how many similar names were flying around in this diff, you're right.